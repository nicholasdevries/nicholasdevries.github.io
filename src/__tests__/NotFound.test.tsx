import { render } from '@testing-library/react'
import { expect, test } from 'vitest'
import NotFound from '../app/not-found'

const { container, getByRole } = render(<NotFound />)

test('has an h1', () => {
  expect(getByRole('heading', { level: 1, name: /404/i })).toBeDefined()
})

test('links home', () => {
  expect(getByRole('link', { name: /return home/i })).toBeDefined()
})

test('matches snapshot', () => {
  expect(container).toMatchSnapshot()
})
