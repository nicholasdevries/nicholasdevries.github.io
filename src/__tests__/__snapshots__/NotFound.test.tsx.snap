// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="text-center md:text-left"
  >
    <h1
      class="text-6xl font-bold tracking-tighter"
    >
      404
    </h1>
    <p
      class="text-2xl tracking-tight text-black/60 dark:text-white/60"
    >
      Page not found
    </p>
    <div
      class="mt-6"
    >
      <a
        class="relative inline-flex h-10 cursor-pointer flex-nowrap items-center justify-center rounded-sm bg-gray-100/50 px-4 font-semibold text-black outline-hidden ring-blue-500 ring-offset-2 ring-offset-white drop-shadow-sm transition-colors hover:bg-gray-100/75 focus-visible:ring-2 dark:bg-gray-900/50 dark:text-white dark:ring-blue-400 dark:ring-offset-gray-800 dark:hover:bg-gray-900/75"
        href="/"
      >
        Return home
      </a>
    </div>
  </div>
</div>
`;
